crates/uv-settings/src/settings.rs

```diff
@@ -136,24 +136,237 @@ pub struct ResolverInstallerOptions {
+    /// directory laid out in the same format.
+    ///
+    /// All indexes provided via this flag take priority over the index specified by
+    /// [`index_url`](#index-url). When multiple indexes are provided, earlier values take priority.
+    #[option(
+        default = "[]",
+        value_type = "list[str]",
+        example = r#"
+            extra-index-url = ["https://download.pytorch.org/whl/cpu"]
+        "#
+    )]
     pub extra_index_url: Option<Vec<IndexUrl>>,
+    /// Ignore all registry indexes (e.g., PyPI), instead relying on direct URL dependencies and
+    /// those provided via `--find-links`.
+    #[option(
+        default = "false",
+        value_type = "bool",
+        example = r#"
+            no-index = true
+        "#
+    )]
     pub no_index: Option<bool>,
+    /// Locations to search for candidate distributions, in addition to those found in the registry
+    /// indexes.
+    ///
+    /// If a path, the target must be a directory that contains packages as wheel files (`.whl`) or
+    /// source distributions (`.tar.gz` or `.zip`) at the top level.
+    ///
+    /// If a URL, the page must contain a flat list of links to package files adhering to the
+    /// formats described above.
+    #[option(
+        default = "[]",
+        value_type = "list[str]",
+        example = r#"
+            find-links = ["https://download.pytorch.org/whl/torch_stable.html"]
+        "#
+    )]
     pub find_links: Option<Vec<FlatIndexLocation>>,
+    /// The strategy to use when resolving against multiple index URLs.
+    ///
+    /// By default, `uv` will stop at the first index on which a given package is available, and
+    /// limit resolutions to those present on that first index (`first-match`). This prevents
+    /// "dependency confusion" attacks, whereby an attack can upload a malicious package under the
+    /// same name to a secondary.
+    ///
+    /// Possible values:
+    ///
+    /// - `"first-index"`:        Only use results from the first index that returns a match for a given package name.
+    /// - `"unsafe-first-match"`: Search for every package name across all indexes, exhausting the versions from the first index before moving on to the next.
+    /// - `"unsafe-best-match"`:  Search for every package name across all indexes, preferring the "best" version found. If a package version is in multiple indexes, only look at the entry for the first index.
+    #[option(
+        default = "\"first-index\"",
+        value_type = "str",
+        example = r#"
+            index-strategy = "unsafe-best-match"
+        "#
+    )]
     pub index_strategy: Option<IndexStrategy>,
+    /// Attempt to use `keyring` for authentication for index URLs.
+    ///
+    /// At present, only `--keyring-provider subprocess` is supported, which configures `uv` to
+    /// use the `keyring` CLI to handle authentication.
+    ///
+    /// Defaults to `disabled`.
```

Comment:
Do we need this if covered in the `option` macro?

Reply:
per https://github.com/astral-sh/uv/pull/5122/files#diff-cd5ab0390b5750a658337173d28e15c9631a156470fb8036633f3fefc4bf96a4R225-R227 this looks redundant